### DISCUSSION TITLE
Extraction of retweet text

### DISCUSSION THREAD

**Discussion by stefan-djurovic** [tweepy/tweepy#1537] [Questions] [answered]
How can I extract the text of a retweeted tweet, I've tried like this: 

```
tweets = api.user_timeline(screen_name=user_id, 
                           count=10,
                           include_rts = True,
                           tweet_mode = 'extended'
                           )
for info in tweets[:10]:
    try:
        if info.retweeted_status:
            print(info.retweeted_status.text)
    except:
        pass
```

**Comment by Harmon758** [comment, marked as answer]
> Retweets can be distinguished from typical Tweets by the existence of a `retweeted_status` attribute.

https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/tweet

This means that Satus/Tweet objects that represent Tweets that are not Retweets will not have a `retweeted_status` attribute.
Instead of using a catch-all try . . . except, you can just handle the `AttributeError` or use [`hasattr`](https://docs.python.org/3/library/functions.html#hasattr).
